site/source/docs/porting/asyncify.rst

```diff
@@ -332,8 +366,25 @@ In this example, a function "func" is called which returns a Number.
       console.log("js_func: " + result);
     });
 
-Optimizing
-##########
+
+Differences Between Asyncify and JSPI
+#####################################
+
+Besides using different underlying mechanisms, Asyncify and JSPI also handle
+async imports and exports differently. Asyncify will automatically determine
+what exports will become async based on what could potentially call an
+an async import (``ASYNCIFY_IMPORTS``). However, with JSPI, the async imports
+and exports must be explicitly set using ``JSPI_IMPORTS`` and ``JSPI_EXPORTS``
+settings.
```

Comment:
Is it not possible to compute exports in the JSPI case so that users don't need to specify them explicitly?

Reply:
It's possible and we started down that path in this [PR](https://github.com/WebAssembly/binaryen/pull/5211). I think the big difference with JSPI, is all "JSPI'ed" exports return a promise even if they don't ever suspend. If we do the auto approach that could lead to exports returning a promise where it's not expected. I think this worked okay with asyncify since often the export didn't actually suspend, so the library code worked as expected.

We could do some hybrid approach, where we check to see what exports we think will suspend and if they're not in the list error out or give the user a list of suspending exports.

Reply:
I see, thanks, makes sense.